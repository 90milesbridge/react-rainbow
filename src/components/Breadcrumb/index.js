/* eslint-disable react/require-default-props */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export default class Breadcrumb extends Component {
    getLiClass() {
        return classnames('slds-breadcrumb__item slds-text-title_caps', this.props.className);
    }

    render() {
        const { href, label, onClick, style } = this.props;

        return (
            <li className={this.getLiClass()} style={style}>
                <a href={href} onClick={onClick}>
                    {label}
                </a>
            </li>
        );
    }
}

Breadcrumb.propTypes = {
    /** The href of the anchor */
    href: PropTypes.string,
    /** Label for the breadcrumb */
    label: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node,
    ]).isRequired,
    /** Callback function fired when the breadcrumb is clicked */
    onClick: PropTypes.func,
    /** Class for custom styles */
    className: PropTypes.string,
    /** Object with the custom styles. The properties must be in camelCase naming
     convention (e.g. { fontFamily: ‘helvetica’ }) */
    style: PropTypes.object,
};

Breadcrumb.defaultProps = {
    href: 'javascript:void(0);', // eslint-disable-line no-script-url
};
