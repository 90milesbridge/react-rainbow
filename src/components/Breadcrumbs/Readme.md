Breadcrumbs examples:

    <Breadcrumbs>
        <Breadcrumb label="Parent entity" onClick={ () => alert('Breadcrumb was clicked') } />
        <Breadcrumb label="Parent record name" />
    </Breadcrumbs>