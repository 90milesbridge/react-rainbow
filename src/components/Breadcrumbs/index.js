/* eslint-disable react/require-default-props */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export default class Breadcrumbs extends Component {
    getContainerClass() {
        return classnames('slds-breadcrumb slds-list_horizontal slds-wrap', this.props.className);
    }

    render() {
        const { children, style } = this.props;
        return (
            <nav aria-label="Breadcrumbs" style={style}>
                <ol className={this.getContainerClass()}>
                    { children }
                </ol>
            </nav>
        );
    }
}

Breadcrumbs.propTypes = {
    /** This is what will be displayed inside the breadcrumbs. It is a list of
     Breadcrumb component childrens */
    children: PropTypes.node,
    /** Class for custom styles */
    className: PropTypes.string,
    /** Object with the custom styles. The properties must be in camelCase naming
     convention (e.g. { fontFamily: ‘helvetica’ }) */
    style: PropTypes.object,
};
