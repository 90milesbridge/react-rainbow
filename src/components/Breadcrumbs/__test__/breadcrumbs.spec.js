import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Breadcrumbs from './../index';
import Breadcrumb from './../../Breadcrumb/index';

describe('<Breadcrumbs/>', () => {
    it('should render a Breadcrumbs with its childrens', () => {
        const component = renderer.create(
            <Breadcrumbs>
                <Breadcrumb label="Parent entity" />
                <Breadcrumb label="Parent record name" />
            </Breadcrumbs>,
        );

        expect(component).toMatchSnapshot();
    });

    it('should have a aria-label attribute for accessibility', () => {
        const component = shallow(
           <Breadcrumbs>
                <Breadcrumb label="Parent entity" />
            </Breadcrumbs>,
        );

        expect(component.find('[aria-label="Breadcrumbs"]').exists()).toBe(true);
    });
});
