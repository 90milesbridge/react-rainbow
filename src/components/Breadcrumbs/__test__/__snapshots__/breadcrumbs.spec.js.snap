// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs/> should render a Breadcrumbs with its childrens 1`] = `
<nav
  aria-label="Breadcrumbs"
  style={undefined}
>
  <ol
    className="slds-breadcrumb slds-list_horizontal slds-wrap"
  >
    <li
      className="slds-breadcrumb__item slds-text-title_caps"
      style={undefined}
    >
      <a
        href="javascript:void(0);"
        onClick={undefined}
      >
        Parent entity
      </a>
    </li>
    <li
      className="slds-breadcrumb__item slds-text-title_caps"
      style={undefined}
    >
      <a
        href="javascript:void(0);"
        onClick={undefined}
      >
        Parent record name
      </a>
    </li>
  </ol>
</nav>
`;
